hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/rsql/RSQLUtilityTest.java

```diff
@@ -99,6 +100,43 @@ TenantConfigurationManagementHolder tenantConfigurationManagementHolder() {
     private static final TenantConfigurationValue<String> TEST_POLLING_OVERDUE_TIME_INTERVAL = TenantConfigurationValue
             .<String> builder().value("00:07:37").build();
 
+    @Test(expected = RSQLParameterUnsupportedFieldException.class)
+    @Description("Testing throwing exception in case of not allowed RSQL key")
+    public void rsqlUnsupportedFieldExceptionTest(){
+        final String rsql = "wrongfield == abcd";
+        RSQLUtility.isValid(rsql);
+    }
+
+    @Test(expected = RSQLParameterUnsupportedFieldException.class)
+    @Description("Testing exception in case of not allowed subkey")
+    public void rsqlUnsupportedSubkeyThrowException(){
+        final String rsql = "ASSIGNEDDS.unsupported == abcd and ASSIGNEDDS.version == 0123";
```

Comment:
I would not stick to the Target fields in this test class as it should be generic. However it makes totally sense to test different behaviour of #isValid method by extending the TestFieldEnum defined at the bottom

Reply:
Using TestFieldEnum now instead of TargetFields as the test object. TestFieldEnum adjusted accordingly.